src/tools/search-variants-mapi.ts

```diff
@@ -0,0 +1,137 @@
+    - Requires language variant filter parameter (e.g., default language '00000000-0000-0000-0000-000000000000')
+    - The tool always RETURNS ONLY TOP 50 most relevant items at max
+    
+    USE FOR:
+    - Conceptual search: Pass single concept (e.g., "neurology", "romance", "technical documentation")
+    - Finding content about topics: Use theme as-is (e.g., "health", "technology")
+    - Content similarity: Find articles similar to a given topic or story theme
+    - Discovering related content when you don't know exact keywords
+    - Thematic content discovery based on meaning and context`,
+    searchOperationSchema.shape,
+    async (
+      { actionName, type, inputs, trackingData },
+      { authInfo: { token, clientId } = {} },
+    ) => {
+      try {
+        const client = createMapiClient(clientId, token, config);
+
+        if (!clientId) {
+          throw new Error("Missing required environment ID");
+        }
+
+        // Step 1: Initiate the AI search operation
+        const searchPayload = {
+          actionName,
+          type,
+          inputs,
+          trackingData,
+        };
+
+        let searchResponse: any;
+        try {
+          searchResponse = await client
+            .post()
+            .withAction(`projects/${clientId}/early-access/ai-operation`)
+            .withData(searchPayload)
+            .toPromise();
+        } catch (error: any) {
+          if (
+            error?.response?.status === 403 &&
+            error?.response?.data?.message?.includes("AI Feature Not Available")
+          ) {
+            return createMcpToolSuccessResponse({
+              status: "unavailable",
+              message:
+                "AI search feature is not available for this environment",
+              reason:
+                "This Kontent.ai project does not have the AI search feature enabled",
+            });
+          }
+          throw error;
+        }
+
+        const operationData: AiOperationResponse = searchResponse.data;
+        const operationId = operationData.operationId;
+
+        // Step 2: Poll for results with exponential backoff
+        const resultData = await pRetry(
+          async () => {
+            const pollResponse = await client
+              .get()
+              .withAction(
+                `projects/${clientId}/early-access/ai-operation/${operationId}`,
+              )
+              .toPromise();
+
+            const data: AiOperationResultResponse = pollResponse.data;
+
+            if (data.message.includes("still in progress")) {
+              throw new Error("Operation still in progress");
+            }
+
+            return data;
+          },
+          {
+            retries: 10,
+            minTimeout: 1000,
+            maxTimeout: 10000,
+            factor: 1.5,
+          },
+        );
+
+        if (resultData.message.includes("completed successfully")) {
+          return createMcpToolSuccessResponse({
```

Comment:
`operationId` should be transparent to the MCP server users (MCP clients) – it should not be here at all. Why do I need both `result` and `message`?